game/src/ui/gameplay/HealthBar/HealthBar.gd

```diff
@@ -0,0 +1,60 @@
+extends Node2D
+
+
+export var init_red_portal_health: int = 100
+export var init_blue_portal_health: int = 100
+onready var red_portal_health: int = init_red_portal_health
+onready var blue_portal_health: int = init_blue_portal_health
+var red_health_display: TextureProgress
+var blue_health_display: TextureProgress
+
+func _ready():
+	red_health_display = self.get_node("Red")
+	blue_health_display = self.get_node("Blue")
+	red_health_display.set_max(init_red_portal_health)
+	blue_health_display.set_max(init_blue_portal_health)
+	update_health()
+	
+func _set_red_portal_health(health: int):
+	red_portal_health = health
+	update_health()
+	
+func _change_red_portal_health(health_change: int):
+	red_portal_health += health_change
+	update_health()
+		
+func _set_blue_portal_health(health: int):
```

Comment:
why is there a set and a change ah? (i think set > change cuz set get is the programmer norm)

Reply:
fixed